crates/fluvio/src/producer/partition_producer.rs

```diff
@@ -212,27 +244,46 @@ impl PartitionProducer {
         request.timeout = self.config.timeout;
         request.topics.push(topic_request);
 
-        let response = spu_socket.send_receive(request).await?;
+        let response = if self.client_stats.is_collect(ClientStatsDataCollect::Data) {
+            let (response, send_latency) = self
+                .client_stats
+                .send_and_measure_latency(&spu_socket, request, total_batch_len)
+                .await?;
+
+            client_stats_update += send_latency;
+            response
+        } else {
+            spu_socket.send_receive(request).await?
+        };
 
+        let mut base_offset = 0;
         for (batch_notifier, partition_response) in batch_notifiers.into_iter().zip(
             response
                 .responses
                 .into_iter()
                 .flat_map(|response| response.partitions),
         ) {
-            let base_offset = partition_response.base_offset;
+            base_offset = partition_response.base_offset;
             let fluvio_error = partition_response.error_code.clone();
+
             if let Err(_e) = batch_notifier
                 .send((base_offset, fluvio_error.clone()))
                 .await
             {
                 trace!("Failed to notify produce result because receiver was dropped");
             }
+
             if fluvio_error.is_error() {
                 return Err(FluvioError::from(ProducerError::from(fluvio_error)));
             }
         }
 
+        // Commit stats update
+        if self.client_stats.is_collect(ClientStatsDataCollect::Data) {
+            client_stats_update.push(ClientStatsMetricRaw::Offset(base_offset as i32));
```

Comment:
This is true, the last offset update is what is what is recorded. The offset in this case is purely for reporting purposes, and not for calculations. I might ask what is your expected behavior for this case? My intent was to reflect the status of the producer after flush is complete.

I haven't tested with multiple partitions yet, so my concern is about whether we accurately capture the volume of traffic sending to all partitions.

Reply:
I'm testing on a 1 node cluster, but I wasn't able to verify that the client sends records to all partitions in a single request using `iftop`

One partition
```
$ fluvio topic create one -p 1
topic "one" created
$ fluvio produce one -f /usr/share/dict/american-english
```

```
TX:             cum:   1.65MB   peak:   6.59Mb      rates:      0b   1.32Mb   422Kb
RX:                    20.4KB           81.7Kb                  0b   16.3Kb  5.10Kb
TOTAL:                 1.67MB           6.67Mb                  0b   1.33Mb   427Kb
```

Ten partitions
```
$ fluvio topic create ten -p 10
topic "ten" created
$ fluvio produce ten -f /usr/share/dict/american-english
```


```
TX:             cum:   1.67MB   peak:   6.68Mb      rates:      0b   1.34Mb  1.34Mb
RX:                    27.9KB            112Kb                  0b   22.3Kb  22.3Kb
TOTAL:                 1.70MB           6.79Mb                  0b   1.36Mb  1.36Mb
```

Reply:
I think we can create a separate issue for multiple partitions so it's not a show stopper for this PR

Reply:
I looked more carefully. There is always one partition in this place because we create one PartitionProducer per partition. So, using the latest offset here is totally valid.